Update Sentry library from v.8.36.0 to v.8.43.0

pmusolino opened the pull request:
## Description
I updated the Sentry library from v8.36.0 to v8.43.0 to check if this upgrade would fix [this issue](https://github.com/getsentry/sentry-cocoa/issues/4728) discussed also here pe5sF9-3IY#comment-4282, but it does not. Anyway, I think it's good to keep it updated.

## Testing information
Sentry logs should work like before.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

pmusolino committed:
update: Sentry library from v.8.36.0 to v.8.43.0

itsmeichigo commented:
Should we go over the changelogs of the versions and see what could affect us/what should be tested?

pmusolino committed:
Merge branch 'trunk' into update/update-sentry-library-from-v8-36-0-to-v-8-43-0

pmusolino commented:
@itsmeichigo I checked all the [changelogs](https://github.com/getsentry/sentry-cocoa/releases?page=2) from 8.36.0 to 8.43.0, and it seems they:
- Added some options for the reports.
- Changes regarding visionOS.
- Experimental feature: transactions for crashes, disabled by default.
- Added in_foreground app context to transactions and crash events.
- Some fixes

In general, I don't see anything that can affect directly WCiOS in wrong ways.

itsmeichigo reviewed (approved):
LGTM :shipit:
